ccan/meson.build

```diff
@@ -6,13 +6,21 @@
 # Authors: Martin Belanger <Martin.Belanger@dell.com>
 #
 
-sources += files([
+sources = [
+    config_h,
```

Comment:
This looks wrong. Why do you need to add the header to the source list?

Reply:
I've seen header files added to sources in other projects (i.e. systemd), but after re-reading the meson documentation there is a better to do this. I will send a fix.

With regards to adding `link_with: libccan` to `libnvme/meson.build`, I don't think that's needed because `libccan.a` is linked to `libnvme.so`. Therefore, the ccan code is part of `libnvme.so`. In `libnvme/meson.build` we link with `libnvme.so` (as shown below), therefore we link the ccan code along with `libnvme.so`.

```
    pynvme_clib = python3.extension_module(
        '_nvme',
        pymod_swig[1],
        dependencies : py3_dep,
        include_directories: [incdir, internal_incdir],
        link_with: libnvme,
        install: true,
        subdir: 'libnvme',
    )
```

Reply:
You need to add the static library to python library. This very idea of the static library started because of the missing ccan symbols in the python library (see #174).

```
7/7 [Python] create ctrl object        FAIL            0.04s   exit status 1
>>> PYTHONPATH=/home/wagi/work/libnvme-upstream/.build/libnvme/.. MALLOC_PERTURB_=0 /usr/bin/python3 /home/wagi/work/libnvme-upstream/.build/../libnvme/tests/create-ctrl-obj.py
―――――――――――――――――――――――――――――――――――――――――――――――――――― ✀  ――――――――――――――――――――――――――――――――――――――――――――――――――――
stderr:
Traceback (most recent call last):
  File "/home/wagi/work/libnvme-upstream/.build/../libnvme/tests/create-ctrl-obj.py", line 4, in <module>
    from libnvme import nvme
  File "/home/wagi/work/libnvme-upstream/.build/libnvme/nvme.py", line 13, in <module>
    from . import _nvme
ImportError: /home/wagi/work/libnvme-upstream/.build/libnvme/_nvme.cpython-38-x86_64-linux-gnu.so: undefined symbol: str_strstr
――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
```

with 
```patch
@@ -31,7 +31,7 @@ if have_python_support
         pymod_swig[1],
         dependencies : py3_dep,
         include_directories: [incdir, internal_incdir],
-        link_with: libnvme,
+        link_with: [libnvme, libccan],
         install: true,
         subdir: 'libnvme',
     )

```

```
[0/1] Running all tests.
1/7 main                               OK              0.01s
2/7 main                               OK              0.01s
3/7 cpp                                OK              0.01s
4/7 register                           OK              0.01s
5/7 zns                                OK              0.01s
6/7 [Python] import libnvme            OK              0.03s
7/7 [Python] create ctrl object        OK              0.03s
```

Reply:
@igaw - Addressed your latest comments. I understand now why you wanted `libccan` explicitly added to `libnvme/meson.build`. It's to solve the missing symbol when `buildtype=debug`. By explicitly linking `libccan` we're basically making all ccan symbols public.

Reply:
The symbols from libccan should not be exported. For libnvme we maintain the libnvme.map. But not for the python library. This needs to be addressed. Ideally in a more usable way than libnvme.map.